## Merging DataFrames Together
Goal of This Notebook:

In this notebook I have two DataFrames with information that I want to analyze. The DataFrame with majority of the information will be my master DataFrame. The second DataFrame only has 3 columns of data that I want and will be the one I merge( ) -or- join( ) to the master DataFrame.

After merging, I will create a barnd new DataFrame that has ALL the information I need in one place. This just makes it easier to clean up.

## Importing the Libraries Needed to Work on DataFrames

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns 
import numpy as np


## Reading My csv Files

Below I'm reading in my csv files to this notebook and assigning each one to its own variable. This way I can refer to the variable instead of the csv file name. You will see this later on in the notebook.

In [3]:
df_num = pd.read_csv('Movies_by_id.csv')

In [4]:
df_movies = pd.read_csv('MostPopularMovies_89_19.csv')

## What's in My DataFrame

Question: Looking at df_movies what can I find out?

Examples:
* Column Names
* Column and Rows Count

## df_movies  Data Overview

In [30]:
# How big is my file?
df_movies.size

99820

In [31]:
# How many rows, colums do I have in my data
df_movies.shape

(7130, 14)

In [41]:
# What are the column names and how many are there?
print(len(df_movies.columns))
print(df_movies.columns)


14
Index(['popularity', 'vote_count', 'video', 'poster_path', 'id', 'adult',
       'backdrop_path', 'original_language', 'original_title', 'genre_ids',
       'title', 'vote_average', 'overview', 'release_date'],
      dtype='object')


In [66]:
# Last 5 rows of this DataFrame
df_movies.tail(30)

popularity  vote_count  video                       poster_path      id  \
7100      12.265        1771  False  /8mBuLCOcpWnmYtZc4aqtvDXslv6.jpg    1621   
7101      12.218        1309  False  /z6yhgEzGUJaBBgAxoyggRzwLnVM.jpg     638   
7102      12.214        1672  False  /fQYgljzKgkeGEukezaJsMWp7oq1.jpg  420814   
7103      12.178        1011  False  /jrzxS0vcbzIIay1sdYm0rgI2QfJ.jpg  458594   
7104      12.170        1736  False  /lDXJUmLQsViSKLbil2luD91l2kE.jpg    1955   
7105      11.956        1199  False  /2L8ehd95eSW9x7KINYtZmRkAlrZ.jpg  347375   
7106      11.890        1287  False  /tpPERI5i6yjT73jYWyZ9iWltQ9L.jpg  454992   
7107      11.805        1325  False  /ryACHOzmRJiJEKjatbokudfkO74.jpg  463272   
7108      11.748        3682  False  /zHJFdhqEXZJxG653oZvOM3PmNON.jpg     565   
7109      11.710        1083  False  /9zUax4pjRgoPIJla9CQejqNJofB.jpg    8913   
7110      11.608        2328  False  /5xNBYXuv8wqiLVDhsfqCOr75DL7.jpg  449563   
7111      11.444        1244  False  /dic3GdmMpxxfkCQfvZnasb5ZkSG.jpg  491418   
7112      11.436        1298  False  /9vPDY8e7YxLwgVum7YZIUJbr4qc.jpg  426543   
7113      11.430        1702  False  /zSuJ3r5zr5T26tTxyygHhgkUAIM.jpg  457799   
7114      11.390        1119  False  /3rViQPcrWthMNecp5XnkKev6BzW.jpg  463684   
7115      11.139        1608  False  /kN9nWy36vyCJyMTRjQPgolxEr8l.jpg   10601   
7116      10.883        2034  False  /vzI4gNJDUwWgmNDSBWxYoeGkIEY.jpg  458423   
7117      10.851        1619  False  /aBw8zYuAljVM1FeK5bZKITPH8ZD.jpg  399361   
7118      10.826        1358  False  /puTDvsUADxCA5NjJuwgEWqQ3xGA.jpg  228326   
7119      10.800        1402  False  /iqWRyStIQikuQkZxEfkss8NqsbS.jpg  267193   
7120      10.737        3266  False  /aukTmf7j4JBFdvqKe4z9kbhEcmw.jpg  201085   
7121      10.668        3919  False  /AhTO2QWG0tug7yDoh0XoaMhPt3J.jpg  346685   
7122      10.455        1122  False  /vlP18ooLmvYVNnqidbot0FupY05.jpg  109410   
7123      10.414        1296  False  /d31SGJSaX29ba5ZUbZcesGoDE7I.jpg  401469   
7124      10.380        2069  False  /kKmeBCe73jZlN1rjfJy1TEZrOw5.jpg  303858   
7125      10.039        1383  False  /afdZAIcAQscziqVtsEoh2PwsYTW.jpg  399360   
7126       9.577        1584  False  /iMkl2Akc1f4CSJCieVwczM4KjZR.jpg  376866   
7127       9.214        1013  False  /szLfTQ2vLFQkNbP3dc0pnco1rje.jpg  323677   
7128       8.289        1405  False  /4zfbf70df2ySChbsIEjhAC9oXJ1.jpg  257088   
7129       1.769        4334  False  /ft8IqAGFs3V7i87z0t0EVRUjK1p.jpg   13804   

      adult                     backdrop_path original_language  \
7100  False  /yV7ty8IThfkQ5hzNwZH1krj86PR.jpg                en   
7101  False  /5UhVmv70qTfy7YSf8r87kNXWxjK.jpg                en   
7102  False  /uDt8bQ4lGVlabEx5Gl2cPzvy6qz.jpg                en   
7103  False  /9krZX6T4EEs5eFpXgSi8jZkN48O.jpg                en   
7104  False  /rR479zn26lJp4vNxXKjfBVCHLAW.jpg                en   
7105  False  /2nOMUnQ4OtsKVTmpp3og7MYt4oG.jpg                en   
7106  False  /3Ptt97kQLuJS37zwA4lkJAkr02m.jpg                en   
7107  False  /b7Q0uX4Tcd6lXWItJEMxYk9XBbc.jpg                en   
7108  False  /kQOLgJGvFjQ7EwGfaPkW9dlGFXJ.jpg                en   
7109  False  /sbm4pRbGNaqaHjx6fPXh1aQGTJO.jpg                en   
7110  False  /fPwJmIg6CVNNXUJvL1wgeexzwEp.jpg                en   
7111  False  /mXowJ0usjsCgI9hM0BSIoTVnKRI.jpg                en   
7112  False  /l2ji4YiNSPBV69WjGBgU0gCvRqy.jpg                en   
7113  False  /uRoA67sM5vvCI4cplxhl67BTPly.jpg                en   
7114  False  /ckNp6LPhp1roR8dVI16q5u3LUMg.jpg                en   
7115  False  /zyWbOvKCdA9V9P5xuUG2eiPYsUo.jpg                en   
7116  False  /8HKwxHZ23sW8wz4yb28IWnAM1M6.jpg                en   
7117  False  /s9I2LmQMYCanl6DvC3X1AOHs2r8.jpg                en   
7118  False  /lVshZpxU5EB8zXJrnXxblaDps6m.jpg                en   
7119  False  /s04T6oIdglrbsZ1WPK2p20J9jag.jpg                en   
7120  False  /zTs5Mc1zfEj61cvolvpPn8zSyim.jpg                en   
7121  False  /hqSS9j

In [27]:
# Statistical Analysis of the votes average, popularity score, movie id, amount of votes 
df_movies.describe()

popularity    vote_count             id  vote_average
count  7130.000000   7130.000000    7130.000000   7130.000000
mean     25.931030   3597.239130  291930.569565      6.789565
std      29.161267   3365.923804  184977.847334      0.774244
min       1.769000   1011.000000      28.000000      4.300000
25%      13.851000   1584.000000  118340.000000      6.200000
50%      18.529500   2483.000000  335890.000000      6.700000
75%      27.318000   4029.000000  449562.000000      7.300000
max     323.546000  19816.000000  567609.000000      8.600000

In [28]:
# What are the data types in my DataFrame and how many of them do I have
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7130 entries, 0 to 7129
Data columns (total 14 columns):
popularity           7130 non-null float64
vote_count           7130 non-null int64
video                7130 non-null bool
poster_path          7130 non-null object
id                   7130 non-null int64
adult                7130 non-null bool
backdrop_path        7130 non-null object
original_language    7130 non-null object
original_title       7130 non-null object
genre_ids            7130 non-null object
title                7130 non-null object
vote_average         7130 non-null float64
overview             7130 non-null object
release_date         7130 non-null object
dtypes: bool(2), float64(2), int64(2), object(8)
memory usage: 682.5+ KB


In [29]:
# Are there any NaN or Nulls aka missing data
df_movies.isna().sum()

popularity           0
vote_count           0
video                0
poster_path          0
id                   0
adult                0
backdrop_path        0
original_language    0
original_title       0
genre_ids            0
title                0
vote_average         0
overview             0
release_date         0
dtype: int64

## df_num Data Overview



In [42]:
# What are the column names and how many are there?
print(len(df_num.columns))
print(df_num.columns)

25
Index(['adult', 'backdrop_path', 'belongs_to_collection', 'budget', 'genres',
       'homepage', 'id', 'imdb_id', 'original_language', 'original_title',
       'overview', 'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')


In [32]:
# How big is my file
df_num.size

91675

In [33]:
# How many rows, colums do I have in my data
df_num.shape

(3667, 25)

In [35]:
# What are the data types in my DataFrame and how many of them do I have
df_num.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3667 entries, 0 to 3666
Data columns (total 25 columns):
adult                    3667 non-null bool
backdrop_path            3667 non-null object
belongs_to_collection    1357 non-null object
budget                   3667 non-null int64
genres                   3667 non-null object
homepage                 2870 non-null object
id                       3667 non-null int64
imdb_id                  3667 non-null object
original_language        3667 non-null object
original_title           3667 non-null object
overview                 3667 non-null object
popularity               3667 non-null float64
poster_path              3667 non-null object
production_companies     3667 non-null object
production_countries     3667 non-null object
release_date             3667 non-null object
revenue                  3667 non-null int64
runtime                  3667 non-null int64
spoken_languages         3667 non-null object
status                  

In [36]:
# Statistical Analysis of the votes average, popularity score, movie id, amounnt of votes 
df_num.describe()

budget             id   popularity       revenue      runtime  \
count  3.667000e+03    3667.000000  3667.000000  3.667000e+03  3667.000000   
mean   6.573071e+07  291875.874830    25.089933  2.622205e+08   117.488956   
std    6.414030e+07  185154.718872    27.209596  3.585256e+08    21.439242   
min    0.000000e+00      28.000000     1.482000  0.000000e+00    83.000000   
25%    1.900000e+07  118340.000000    14.201000  4.891904e+07   102.000000   
50%    4.500000e+07  335983.000000    18.385000  1.379443e+08   116.000000   
75%    1.000000e+08  449562.000000    25.949000  3.278037e+08   128.000000   
max    3.560000e+08  567609.000000   272.710000  2.797801e+09   209.000000   

       vote_average    vote_count  
count   3667.000000   3667.000000  
mean       6.791437   3606.173711  
std        0.774511   3372.108794  
min        4.300000   1011.000000  
25%        6.200000   1585.000000  
50%        6.700000   2492.000000  
75%        7.300000   4029.000000  
max        8.600000  19823.000000

In [38]:
# Are there any NaN or Nulls aka missing data
df_num.isna().sum()

adult                       0
backdrop_path               0
belongs_to_collection    2310
budget                      0
genres                      0
homepage                  797
id                          0
imdb_id                     0
original_language           0
original_title              0
overview                    0
popularity                  0
poster_path                 0
production_companies        0
production_countries        0
release_date                0
revenue                     0
runtime                     0
spoken_languages            0
status                      0
tagline                   271
title                       0
video                       0
vote_average                0
vote_count                  0
dtype: int64

In [45]:
# First five rows of the DataFrame
df_num.head()

adult                     backdrop_path  \
0  False  /cjTQSwcsfVdirSFSHNBXRGkxmWa.jpg   
1  False  /n6bUvigpRFqSwmPp1m2YADdbRBc.jpg   
2  False  /riTANvQ8GKmQbgtC1ps3OfkU43A.jpg   
3  False  /bOGkgRGdhrBYJSLpXaxhXVstddV.jpg   
4  False  /cqa3sa4c4jevgnEJwq3CMF8UfTG.jpg   

                               belongs_to_collection     budget  \
0                                                NaN   40000000   
1                                                NaN   55000000   
2  {'id': 528, 'name': 'The Terminator Collection...  185000000   
3  {'id': 86311, 'name': 'The Avengers Collection...  300000000   
4                                                NaN  100000000   

                                              genres  \
0  [{'id': 9648, 'name': 'Mystery'}, {'id': 53, '...   
1  [{'id': 80, 'name': 'Crime'}, {'id': 53, 'name...   
2  [{'id': 28, 'name': 'Action'}, {'id': 878, 'na...   
3  [{'id': 12, 'name': 'Adventure'}, {'id': 28, '...   
4  [{'id': 10752, 'name': 'War'}, {'id': 18, 'nam...   

                                            homepage      id    imdb_id  \
0                       https://www.knivesout.movie/  546554  tt8946378   
1                         http://www.jokermovie.net/  475557  tt7286456   
2  https://www.paramountmovies.com/movies/termina...  290859  tt6450804   
3  https://www.marvel.com/movies/avengers-infinit...  299536  tt4154756   
4                            https://www.1917.movie/  530915  tt8579674   

  original_language          original_title  ... release_date     revenue  \
0                en              Knives Out  ...   2019-11-27   163700000   
1                en                   Joker  ...   2019-10-02  1073698955   
2                en   Terminator: Dark Fate  ...   2019-10-23   233685077   
3                en  Avengers: Infinity War  ...   2018-04-25  2046239637   
4                en                    1917  ...   2019-12-10   200483309   

  runtime                                   spoken_languages    status  \
0     131  [{'iso_639_1': 'es', 'name': 'Español'}, {'iso...  Released   
1     122           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
2     128  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
3     149  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
4     119  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   

                                  tagline                   title  video  \
0   Hell, any of them could have done it.              Knives Out  False   
1                    Put on a happy face.                   Joker  False   
2  Welcome to the day after judgement day   Terminator: Dark Fate  False   
3   An entire universe. Once and for all.  Avengers: Infinity War  False   
4                       Time is the enemy                    1917  False   

  vote_average vote_count  
0          7.8       2488  
1          8.2       9552  
2          6.3       1580  
3          8.3      16900  
4          8.0       2974  

[5 rows x 25 columns]

## What Do You See

For starters, when looking at the data frames I will not be able to concate() them together when it comes to merging. This is because the column sizes do not match. What I plan on doing is merging -or- joining (I say or because I have to look at the documentation for both methods before I know which one I should use) the two DataFrames together first to create a new DataFrame that has all of my data. After doing so I want to then clean my data up i.e. dropping repeats.

## Plan 
What I plan on doing is merging -or- joining (I say or because I have to look at the documentation for both methods before I know which one I should use) the two DataFrames together first to create a new DataFrame that has all of my data. 

After doing so I want to then clean my data up(This will be shown in another notebook).
 

## Executing the Plan

In [53]:
# Here I am merging my two DataFrsmes ans assigning it to a variable I can refer to later in the notebook
df_merge = pd.merge(df_movies, df_num, how='outer')

## Checking My Work


In [59]:
# Here I am looking at the column names to see if the 
# columns from both DataFrames are present not including the columns they shared

df_merge.columns

Index(['popularity', 'vote_count', 'video', 'poster_path', 'id', 'adult',
       'backdrop_path', 'original_language', 'original_title', 'genre_ids',
       'title', 'vote_average', 'overview', 'release_date',
       'belongs_to_collection', 'budget', 'genres', 'homepage', 'imdb_id',
       'production_companies', 'production_countries', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline'],
      dtype='object')

## An Issue Has Been Brought to My Attention


The merge worked! But Something isn't right....
The data I thought I was going to have had over 10,000 rows BUT they movies in the rows were being repeated!! 
So I have to drop the movies that keep repeating in my DataFrame. At the end of this I should have the true size and shape of my newly merged DataFrame.

Here goes nothing:

In [28]:
# When I run this I see more than 1 row that has this movie id! 
# I do not want that because it will mess up my analysis of the data

df_merge.loc[df_merge['id'] == 546554].shape

(47, 26)

## Drop Repeats

In [85]:
# The repeats will be dropped here  
# I will assign this to a variable
# This is the first step I have taken to clean my data. 
df_new_merged = df_merge.drop_duplicates(keep= 'first')

## df_new_merged Overview

In [86]:
# First five rows of the data
df_new_merged.head()

popularity  vote_count  video                       poster_path      id  \
0       323.546        2478  False  /xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg  419704   
31      233.733        1864  False  /bB42KDdfWkOvmzmYkmK58ZlCa9P.jpg  512200   
62      127.227        2591  False  /pjeMs3yqRmFL3giJy4PMXWZTTPa.jpg  330457   
93      108.054        2469  False  /pThyQovXQrw2m0s9x82twj48Jq4.jpg  546554   
124      98.211        9521  False  /udDclJoHjfjb8Ekgsd4FDteOkCU.jpg  475557   

     adult                     backdrop_path original_language  \
0    False  /5BwqwxMEjeFtdknRV792Svo0K1v.jpg                en   
31   False  /hreiLoPysWG79TsyQgMzFKaOTF5.jpg                en   
62   False  /xJWPZIYOEFIjZpBL7SVBGnzRYXp.jpg                en   
93   False  /cjTQSwcsfVdirSFSHNBXRGkxmWa.jpg                en   
124  False  /n6bUvigpRFqSwmPp1m2YADdbRBc.jpg                en   

              original_title                genre_ids  ... genres  homepage  \
0                   Ad Astra  [12, 18, 9648, 878, 53]  ...    NaN       NaN   
31   Jumanji: The Next Level         [28, 12, 35, 14]  ...    NaN       NaN   
62                 Frozen II          [12, 16, 10751]  ...    NaN       NaN   
93                Knives Out   [35, 80, 18, 9648, 53]  ...    NaN       NaN   
124                    Joker             [80, 18, 53]  ...    NaN       NaN   

    imdb_id production_companies production_countries  revenue runtime  \
0       NaN                  NaN                  NaN      NaN     NaN   
31      NaN                  NaN                  NaN      NaN     NaN   
62      NaN                  NaN                  NaN      NaN     NaN   
93      NaN                  NaN                  NaN      NaN     NaN   
124     NaN                  NaN                  NaN      NaN     NaN   

    spoken_languages status tagline  
0                NaN    NaN     NaN  
31               NaN    NaN     NaN  
62               NaN    NaN     NaN  
93               NaN    NaN     NaN  
124              NaN    NaN     NaN  

[5 rows x 26 columns]

In [87]:
# Last five rows of the data.
# It look like the df_num was added to the bottom of df_movies
df_new_merged.tail()

popularity  vote_count  video                       poster_path  \
10722      11.605        1405  False  /4zfbf70df2ySChbsIEjhAC9oXJ1.jpg   
10737       1.482        4337  False  /ft8IqAGFs3V7i87z0t0EVRUjK1p.jpg   
10752     271.489        2486  False  /xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg   
10767     272.710        1916  False  /bB42KDdfWkOvmzmYkmK58ZlCa9P.jpg   
10782     124.257        2618  False  /pjeMs3yqRmFL3giJy4PMXWZTTPa.jpg   

           id  adult                     backdrop_path original_language  \
10722  257088  False  /x2JJGVftWfx2JGyqW6pqP3JrkO0.jpg                en   
10737   13804  False  /1R2ihQztuRTIqN3pFBR1yrBMd7w.jpg                en   
10752  419704  False  /5BwqwxMEjeFtdknRV792Svo0K1v.jpg                en   
10767  512200  False  /hreiLoPysWG79TsyQgMzFKaOTF5.jpg                en   
10782  330457  False  /xJWPZIYOEFIjZpBL7SVBGnzRYXp.jpg                en   

                original_title genre_ids  ...  \
10722              Point Break       NaN  ...   
10737           Fast & Furious       NaN  ...   
10752                 Ad Astra       NaN  ...   
10767  Jumanji: The Next Level       NaN  ...   
10782                Frozen II       NaN  ...   

                                                  genres  \
10722  [{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...   
10737  [{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...   
10752  [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
10767  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
10782  [{'id': 16, 'name': 'Animation'}, {'id': 10751...   

                                        homepage    imdb_id  \
10722   https://www.facebook.com/PointBreakMovie  tt2058673   
10737         http://www.fastandfuriousmovie.net  tt1013752   
10752  https://www.foxmovies.com/movies/ad-astra  tt2935510   
10767                    http://jumanjimovie.com  tt7975244   
10782         https://movies.disney.com/frozen-2  tt4520988   

                                    production_companies  \
10722  [{'id': 264, 'logo_path': '/fA90qwUKgPhMONqtwY...   
10737  [{'id': 1778, 'logo_path': '/69xm6sF3JxugGVjfZ...   
10752  [{'id': 490, 'logo_path': None, 'name': 'New R...   
10767  [{'id': 5, 'logo_path': '/71BqEFAF4V3qjjMPCpLu...   
10782  [{'id': 6125, 'logo_path': '/tVPmo07IHhBs4Huil...   

                                    production_countries       revenue  \
10722  [{'iso_3166_1': 'DE', 'name': 'Germany'}, {'is...  1.337187e+08   
10737  [{'iso_3166_1': 'US', 'name': 'United States o...  3.631643e+08   
10752  [{'iso_3166_1': 'BR', 'name': 'Brazil'}, {'iso...  1.271759e+08   
10767  [{'iso_3166_1': 'US', 'name': 'United States o...  3.108300e+08   
10782  [{'iso_3166_1': 'US', 'name': 'United States o...  1.330765e+09   

      runtime                                   spoken_languages    status  \
10722   114.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
10737   107.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
10752   123.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
10767   123.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
10782   104.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                              tagline  
10722            The only law that matters is gravity  
10737                      New Model. Original Parts.  
10752  The answers we seek are just outside our reach  
10767                                             NaN  
10782                  The past is not what it seems.  

[5 rows x 26 columns]

In [88]:
# How big is my file
df_new_merged.size

11960

In [89]:
# How many rows, columns does the data have 
df_new_merged.shape

(460, 26)

In [90]:
# What are the data types, how many of them do I have
df_new_merged.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 460 entries, 0 to 10782
Data columns (total 26 columns):
popularity               460 non-null float64
vote_count               460 non-null int64
video                    460 non-null object
poster_path              460 non-null object
id                       460 non-null int64
adult                    460 non-null object
backdrop_path            460 non-null object
original_language        460 non-null object
original_title           460 non-null object
genre_ids                230 non-null object
title                    460 non-null object
vote_average             460 non-null float64
overview                 460 non-null object
release_date             460 non-null object
belongs_to_collection    85 non-null object
budget                   230 non-null float64
genres                   230 non-null object
homepage                 180 non-null object
imdb_id                  230 non-null object
production_companies     230 non-null 

In [92]:
# What are the names of the columns and how many columns are there
print(len(df_new_merged.columns))
df_new_merged.columns

26


Index(['popularity', 'vote_count', 'video', 'poster_path', 'id', 'adult',
       'backdrop_path', 'original_language', 'original_title', 'genre_ids',
       'title', 'vote_average', 'overview', 'release_date',
       'belongs_to_collection', 'budget', 'genres', 'homepage', 'imdb_id',
       'production_companies', 'production_countries', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline'],
      dtype='object')

## Conclusion 

Unfortunatly this is not enough data for me to use for my project. 
Why did my data go from over 7,000 rows to 460? 
Well, when pulling my data I failed to realize that I was grabbing data based off of year for my time frame vs the release date

Ex: Batman comes out in 2010 right, but a group of people have voted this as their favorite movie for the past 10 years. My API criteria doesn't know that this is happening and just pulls from 2010 to 2020 the movies that were voted the most popular for the every year during a 10yr time frame. That will mean if enough people voted for Batman as their favorite movie in the time frame I chose, the movie will repeat itself for more than one year. Now imagine this happening for more than one movie over a 10yr span.

This is why I had so much data in the beginning amd so little data at the end. 

## Good News

Good News is this notebook provides examples of how to get an overview of the data in your DataFrame.
You were able to see how to locate and drop repeats in your DataFrame, which is needed when cleaning "messy" data. Located below are the links I used to look at and manipulate/clean my data. Hope it helps, Happy Coding!!!!

## Resources



## Merging / Concatenate / Join:  

https://pandas.pydata.org/pandas-docs/stable/user_guide/merging.html


https://www.datacamp.com/community/tutorials/joining-dataframes-pandas


## How to locate a column in a DataFrame:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.loc.html?highlight=loc#pandas.DataFrame.loc


## Finding Duplicates:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.duplicated.html?highlight=duplicates


## Droping Duplicates:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop.html?highlight=drop#pandas.DataFrame.drop
